src/main/java/com/example/hibernate/service/ModelService.java

```diff
@@ -31,14 +32,14 @@ public Model getModelById(UUID id) throws Exception{
             return optModel.get();
     }
 
-    public Model addModel(Model model)
+    public Model addModel(final AddModel addModel)
     {
-        return modelRepository.save(model);
+        return modelRepository.save(new Model(addModel.name(), addModel.parts()));
     }
 
     public Model updateModel(@PathVariable("id") UUID id, @RequestBody Model model){
```

Comment:
@PathVariable and @RequestBody make sense ONLY in controller. Remove them here asap